services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -469,6 +468,197 @@ public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String
+
+        return updateRoutingRuleResponse;
+    }
+
+    @Override
+    public InternalRouterInfo constructInternalRouterInfo (List<InternalSubnetRoutingTable> internalSubnetRoutingTableList) {
+
+        String requestId = UUID.randomUUID().toString();
+        InternalRouterInfo internalRouterInfo = new InternalRouterInfo();
+
+        InternalRouterConfiguration internalRouterConfiguration = new InternalRouterConfiguration();
+
+        internalRouterConfiguration.setSubnetRoutingTables(internalSubnetRoutingTableList);
+
+        internalRouterConfiguration.setRevisionNumber(ConstantsConfig.REVISION_NUMBER);
+        internalRouterConfiguration.setFormatVersion(ConstantsConfig.FORMAT_VERSION);
+        internalRouterConfiguration.setRequestId(requestId);
+        internalRouterConfiguration.setHostDvrMac("");
+        internalRouterConfiguration.setMessageType(MessageType.FULL);
+
+        internalRouterInfo.setOperationType(OperationType.INFO);
+        internalRouterInfo.setRouterConfiguration(internalRouterConfiguration);
+
+        return internalRouterInfo;
+
+    }
+
+    private InternalRoutingRule constructNewInternalRoutingRule(OperationType operationType, RoutingRuleType routingRuleType, RouteEntry route, NewRoutesRequest newRouteRequest) {
+        if (route == null && newRouteRequest == null) {
+            return new InternalRoutingRule();
+        }
+
+        InternalRoutingRule internalRoutingRule = new InternalRoutingRule();
+
+        InternalRoutingRuleExtraInfo routingRuleExtraInfo = new InternalRoutingRuleExtraInfo();
+        //routingRuleExtraInfo.setNextHopMac();
+        // TODO: insert destination type - if it is vpc router, configure value according to target
+        routingRuleExtraInfo.setDestinationType(VpcRouteTarget.LOCAL);
+
+        if (route == null) {
+            internalRoutingRule.setId(route.getId());
+        } else {
+            internalRoutingRule.setId(UUID.randomUUID().toString());
+            internalRoutingRule.setName(route.getName());
+        }
+
+        if (newRouteRequest != null) {
+            internalRoutingRule.setDestination(newRouteRequest.getDestination());
+        } else {
+            internalRoutingRule.setDestination(route.getDestination());
+        }
+        // TODO: translate target to nextHop - it is vpc router operation
+        if (newRouteRequest != null) {
+            internalRoutingRule.setNextHopIp(newRouteRequest.getNexthop());
+        } else {
+            internalRoutingRule.setNextHopIp(route.getNexthop());
+        }
+        // TODO: set priority - configure priority according to RoutingRuleType
+        if (routingRuleType.getRoutingRuleType().equals(ConstantsConfig.DEFAULT_ROUTINGRULETYPE)) {
+            internalRoutingRule.setPriority(ConstantsConfig.LOW_PRIORITY);
+        } else if (routingRuleType.getRoutingRuleType().equals(ConstantsConfig.STATIC_ROUTINGRULETYPE)) {
+            internalRoutingRule.setPriority(ConstantsConfig.HIGH_PRIORITY);
```

Comment:
`priority` property is an integer, how do you plan to convert these string to integer?

Reply:
It is a String actually ~